tests/Unit/CrudPanel/CrudPanelButtonsTest.php

```diff
@@ -0,0 +1,218 @@
+
+        $this->assertEquals($expectedButton, $this->crudPanel->buttons->last());
+    }
+
+    public function testAddButtonFromView()
+    {
+        $expectedButton = $this->topViewButton;
+        $viewName = 'someViewName';
+
+        $this->crudPanel->addButtonFromView($expectedButton->stack, $expectedButton->name, $viewName, $expectedButton->content);
+
+        $backpackButtonViewPackage = 'vendor.backpack.crud.buttons.';
+        $actualButton = $this->crudPanel->buttons->last();
+
+        $this->assertEquals($expectedButton->stack, $actualButton->stack);
+        $this->assertEquals($expectedButton->name, $actualButton->name);
+        $this->assertEquals($backpackButtonViewPackage.$viewName, $actualButton->content);
+        $this->assertEquals($expectedButton->stack, $actualButton->stack);
+    }
+
+    public function testRemoveButton()
+    {
+        $buttonName = $this->defaultButtonNames[0];
+
+        $this->crudPanel->removeButton($buttonName);
+
+        $this->assertEquals(count($this->defaultButtonNames) - 1, count($this->crudPanel->buttons));
+        $this->assertNull($this->getButtonByName($buttonName));
+    }
+
+    public function testRemoveUnknownButton()
+    {
+        $buttonName = 'someButtonName';
+
+        $this->crudPanel->removeButton($buttonName);
+
+        $this->assertEquals(count($this->defaultButtonNames), count($this->crudPanel->buttons));
+    }
+
+    public function testRemoveAllButtons()
+    {
+        $this->crudPanel->removeAllButtons();
+
+        $this->assertEmpty($this->crudPanel->buttons);
+    }
+
+    public function testRemoveButtonFromStack()
+    {
+        $button = $this->crudPanel->buttons->first();
+
+        $this->crudPanel->removeButtonFromStack($button->name, $button->stack);
+
+        $this->assertEquals(count($this->defaultButtonNames) - 1, count($this->crudPanel->buttons));
+        $this->assertNull($this->getButtonByName($button->name));
+    }
+
+    public function testRemoveUnknownButtonFromStack()
+    {
+        $this->crudPanel->removeButtonFromStack('someButtonName', 'line');
+
+        $this->assertEquals(count($this->defaultButtonNames), count($this->crudPanel->buttons));
+    }
+
+    public function testRemoveButtonFromUnknownStack()
+    {
+        $button = $this->crudPanel->buttons->first();
+
+        $this->crudPanel->removeButtonFromStack($button->name, 'someStackName');
+
+        $this->assertEquals(count($this->defaultButtonNames), count($this->crudPanel->buttons));
+    }
+
+    public function testRemoveAllButtonsFromStack()
+    {
+        $this->crudPanel->removeAllButtonsFromStack('line');
+
+        $this->assertEquals(2, count($this->crudPanel->buttons));
+    }
+
+    public function testRemoveAllButtonsFromUnknownStack()
+    {
+        $this->crudPanel->removeAllButtonsFromStack('someStackName');
+
+        $this->assertEquals(count($this->defaultButtonNames), count($this->crudPanel->buttons));
```

Comment:
And check if there are none/the expected number there?

One probably should somehow check that the other stacks weren't changed _at all_?

Reply:
Yup, will check for equality as well.